BWEM/src/area.cpp

```diff
@@ -414,6 +413,138 @@ void Area::CreateBases()
 	}
 }
 
+// Thanks to Dave Churchill and the SAIDA team
+void Area::calcBoundaryVertices() const
+{
+	set<Position> unsortedVertices;
+
+	for (auto x(0); x < Map::Instance().Size().x; ++x)
+		for (auto y(0); y < Map::Instance().Size().y; ++y)
+		{
+			TilePosition tp{ x, y };
+
+			if (Map::Instance().GetArea(tp) != this)
+			{
+				continue;
+			}
+
+			TilePosition t1{ tp.x + 1, tp.y };
+			TilePosition t2{ tp.x, tp.y + 1 };
+			TilePosition t3{ tp.x - 1, tp.y };
+			TilePosition t4{ tp.x, tp.y - 1 };
+
+			// a tile is 'surrounded' if
+			// 1) in all 4 directions there's a tile position in the current region
+			// 2) in all 4 directions there's a buildable tile
+			auto surrounded = true;
+
+			if (tp.x == 0 || tp.y == 0
```

Comment:
Should we also check bottom/right edges here and not just left/top?

Reply:
Done